src/Microsoft.ML.ImageAnalytics/ImageLoader.cs

```diff
@@ -196,7 +196,14 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
                                 string path = src.ToString();
                                 if (!string.IsNullOrWhiteSpace(_parent.ImageFolder))
                                     path = Path.Combine(_parent.ImageFolder, path);
-                                dst = new Bitmap(path);
+
+                                MemoryStream inMemoryCopy = new MemoryStream();
+                                using (FileStream fs = File.OpenRead(path))
+                                {
+                                    fs.CopyTo(inMemoryCopy);
+                                }
```

Comment:
Probably simpler as `var inMemoryCopy = new MemoryStream(File.ReadAllBytes());`.

Reply:
Would also be nice if it had some protection again, say, being pointed to a huge non-image file.

---
In reply to: [280130176](https://github.com/dotnet/machinelearning/pull/3625#discussion_r280130176) [](ancestors = 280130176)